packages/storefront-lwc/src/modules/commerce/headerCart/headerCart.js

```diff
@@ -5,10 +5,28 @@
     For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
 import { LightningElement, track } from 'lwc';
+import { ShoppingCart } from 'commerce/data';
 
 /**
  * Header cart component that should show up in the header
  */
 export default class HeaderCart extends LightningElement {
     @track quantity = 0;
+
+    constructor() {
+        super();
+        ShoppingCart.updateCartListener(this.updateCartHandler.bind(this));
```

Comment:
This isn't a good pattern for using the shopping cart data. One problem in particular is the call to `ShoppingCart.updateCartListener()` actually executes `this.listeners.push(callback);`. This means there is a memory leak, because if this component ever gets unmounted then remounted, it will keep adding new listeners, never cleaning up the old listener.

Also, this pattern isn't very LWC or component friendly. Instead of having the state exist within an external singleton. We should consider using a wire-adapter and/or putting it in a component higher up the hierarchy.